File: core/src/main/java/dev/kkorolyov/pancake/core/component/media/Graphic.java
package dev.kkorolyov.pancake.core.component.media;

import dev.kkorolyov.pancake.platform.entity.Component;
import dev.kkorolyov.pancake.platform.media.graphic.RenderTransform;
import dev.kkorolyov.pancake.platform.media.graphic.Renderable;

/**
 * A swappable visual representation.
 */
public class Graphic implements Component, Renderable {
	private Renderable delegate;

	/**
	 * Constructs a new graphic.
	 * @param delegate delegate providing rendering logic
	 */
	public Graphic(Renderable delegate) {
		setDelegate(delegate);
	}

	/** @return delegate providing rendering logic */
	public Renderable getDelegate() {
		return delegate;
	}
	/**
	 * @param delegate delegate providing rendering logic
	 * @return {@code this}
	 */
	public Graphic setDelegate(Renderable delegate) {
		this.delegate = delegate;
		return this;
	}

	@Override
	public void render(RenderTransform transform) {
		delegate.render(transform);
	}
}

File: graphics-jfx/src/main/kotlin/dev/kkorolyov/pancake/graphics/jfx/component/Graphic.kt
package dev.kkorolyov.pancake.graphics.jfx.component

import dev.kkorolyov.pancake.graphics.jfx.drawable.Drawable
import dev.kkorolyov.pancake.platform.entity.Component

/**
 * A [Drawable] component.
 */
class Graphic(private val delegate: Drawable) : Component, Drawable by delegate
